File: backend/src/models/Restaurant.js
const mongoose = require('mongoose');

const openingHoursSchema = new mongoose.Schema(
  {
    day: { type: String, enum: ['monday','tuesday','wednesday','thursday','friday','saturday','sunday'] },
    open: String,
    close: String,
    isClosed: { type: Boolean, default: false },
  },
  { _id: false }
);

const restaurantSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true, maxlength: 200 },
    description: { type: String, trim: true, maxlength: 2000 },
    address: { type: String, required: true, trim: true },
    city: { type: String, required: true, trim: true },
    postalCode: { type: String, trim: true },
    country: { type: String, default: 'Netherlands', trim: true },
    imageUrl: String,
    cuisineType: [{ type: String, trim: true }],
    openingHours: [openingHoursSchema],
    phoneNumber: String,
    website: String,
    isActive: { type: Boolean, default: true },
    createdBy: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  },
  {
    timestamps: true,
    toJSON: { virtuals: true },
  }
);

restaurantSchema.index({ name: 'text', city: 'text', description: 'text' });
restaurantSchema.index({ city: 1 });
restaurantSchema.index({ cuisineType: 1 });

module.exports = mongoose.model('Restaurant', restaurantSchema);

File: android/app/src/main/java/com/sportsnutrition/domain/models/Restaurant.kt
package com.sportsnutrition.domain.models

data class Restaurant(
    val id: String,
    val name: String,
    val description: String? = null,
    val address: String,
    val city: String,
    val postalCode: String? = null,
    val imageUrl: String? = null,
    val cuisineType: List<String> = emptyList(),
    val openingHours: List<OpeningHours> = emptyList(),
    val phoneNumber: String? = null,
    val website: String? = null,
)

data class OpeningHours(
    val day: String,
    val open: String,
    val close: String,
    val isClosed: Boolean = false,
)
